<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <author>Geoff Lawler</author>
 <class>messageStreamPlaybackForm</class>
 <widget class="QWidget" name="messageStreamPlaybackForm">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>651</width>
    <height>69</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="0" column="0">
    <layout class="QVBoxLayout" name="verticalLayout">
     <property name="spacing">
      <number>6</number>
     </property>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout" stretch="0,0,0,0,0,0,0,0,0">
       <property name="sizeConstraint">
        <enum>QLayout::SetMinimumSize</enum>
       </property>
       <item>
        <widget class="QPushButton" name="rewindToStartButton">
         <property name="toolTip">
          <string>&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'Sans Serif'; font-size:10pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;Rewind to start&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="statusTip">
          <string>&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;\np, li { white-space: pre-wrap; }\n&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'Sans Serif'; font-size:10pt; font-weight:400; font-style:normal;&quot;&gt;\n&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;Rewind to start&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="whatsThis">
          <string>&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'Sans Serif'; font-size:10pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;Go to start of playback&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="text">
          <string/>
         </property>
         <property name="icon">
          <iconset resource="../data/qtResources/watcherResources.qrc">
           <normaloff>:/icons/resources/gtk-media-next-rtl.png</normaloff>:/icons/resources/gtk-media-next-rtl.png</iconset>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="rewindButton">
         <property name="toolTip">
          <string>View stream in reverse direction at current rate  (or click again to double current rate)</string>
         </property>
         <property name="statusTip">
          <string>View stream in reverse direction at current rate  (or click again to double current rate)</string>
         </property>
         <property name="whatsThis">
          <string>Go backward in time at current rate</string>
         </property>
         <property name="text">
          <string/>
         </property>
         <property name="icon">
          <iconset resource="../data/qtResources/watcherResources.qrc">
           <normaloff>:/icons/resources/gtk-media-forward-rtl.png</normaloff>:/icons/resources/gtk-media-forward-rtl.png</iconset>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="playButton">
         <property name="toolTip">
          <string>Play in forward direction at normal rate</string>
         </property>
         <property name="statusTip">
          <string>Play in forward direction at normal rate</string>
         </property>
         <property name="whatsThis">
          <string>Play in forward direction at normal rate</string>
         </property>
         <property name="text">
          <string/>
         </property>
         <property name="icon">
          <iconset resource="../data/qtResources/watcherResources.qrc">
           <normaloff>:/icons/resources/gtk-media-play-ltr.png</normaloff>:/icons/resources/gtk-media-play-ltr.png</iconset>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="pauseButton">
         <property name="toolTip">
          <string>Pause/Unpause playback</string>
         </property>
         <property name="statusTip">
          <string>Pause/Unpause playback</string>
         </property>
         <property name="whatsThis">
          <string>Pause/Unpause playback</string>
         </property>
         <property name="text">
          <string/>
         </property>
         <property name="icon">
          <iconset resource="../data/qtResources/watcherResources.qrc">
           <normaloff>:/icons/resources/gtk-media-pause.png</normaloff>:/icons/resources/gtk-media-pause.png</iconset>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="forwardButton">
         <property name="toolTip">
          <string>View stream in forward direction at current rate (or click again to double current rate)</string>
         </property>
         <property name="statusTip">
          <string>View stream in forward direction at current rate (or click again to double current rate)</string>
         </property>
         <property name="whatsThis">
          <string>Go forward in time at current rate</string>
         </property>
         <property name="text">
          <string/>
         </property>
         <property name="icon">
          <iconset resource="../data/qtResources/watcherResources.qrc">
           <normaloff>:/icons/resources/gtk-media-forward-ltr.png</normaloff>:/icons/resources/gtk-media-forward-ltr.png</iconset>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="forwardToEndButton">
         <property name="toolTip">
          <string>&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'Sans Serif'; font-size:10pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;Forward to end&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="statusTip">
          <string>&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;\np, li { white-space: pre-wrap; }\n&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'Sans Serif'; font-size:10pt; font-weight:400; font-style:normal;&quot;&gt;\n&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;Forward to end&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="whatsThis">
          <string>&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'Sans Serif'; font-size:10pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;Forward to end of playback (go to live view)&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="text">
          <string/>
         </property>
         <property name="icon">
          <iconset resource="../data/qtResources/watcherResources.qrc">
           <normaloff>:/icons/resources/gtk-media-previous-rtl.png</normaloff>:/icons/resources/gtk-media-previous-rtl.png</iconset>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QDoubleSpinBox" name="streamRateSpinBox">
         <property name="toolTip">
          <string>Set playback rate</string>
         </property>
         <property name="whatsThis">
          <string>Set playback rate</string>
         </property>
         <property name="accelerated">
          <bool>true</bool>
         </property>
         <property name="correctionMode">
          <enum>QAbstractSpinBox::CorrectToNearestValue</enum>
         </property>
         <property name="keyboardTracking">
          <bool>false</bool>
         </property>
         <property name="minimum">
          <double>-100.000000000000000</double>
         </property>
         <property name="maximum">
          <double>100.000000000000000</double>
         </property>
         <property name="singleStep">
          <double>0.500000000000000</double>
         </property>
         <property name="value">
          <double>1.000000000000000</double>
         </property>
        </widget>
       </item>
       <item>
        <spacer>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QLabel" name="playbackTimeLabel">
         <property name="text">
          <string>Current Playback Time</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <widget class="QSlider" name="playbackSlider">
       <property name="enabled">
        <bool>true</bool>
       </property>
       <property name="toolTip">
        <string>Drag and drop to desired playback location</string>
       </property>
       <property name="statusTip">
        <string>Drag and drop to desired playback location</string>
       </property>
       <property name="whatsThis">
        <string>Drag and drop to desired playback location</string>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="invertedAppearance">
        <bool>false</bool>
       </property>
       <property name="invertedControls">
        <bool>false</bool>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources>
  <include location="../data/qtResources/watcherResources.qrc"/>
 </resources>
 <connections/>
 <designerdata>
  <property name="gridDeltaX">
   <number>10</number>
  </property>
  <property name="gridDeltaY">
   <number>10</number>
  </property>
  <property name="gridSnapX">
   <bool>true</bool>
  </property>
  <property name="gridSnapY">
   <bool>true</bool>
  </property>
  <property name="gridVisible">
   <bool>true</bool>
  </property>
 </designerdata>
 <slots>
  <signal>playbackPaused()</signal>
  <signal>playbackNormal()</signal>
  <signal>playbackReversed()</signal>
  <signal>playbackForward()</signal>
  <signal>rewoundToStartOfPlayback()</signal>
  <signal>forwardedToEndOfPlayback()</signal>
  <signal>streamRateSet(double)</signal>
  <signal>currentTimestampUpdated(watcher::Timestamp)</signal>
  <signal>messageStreamDirectionChange()</signal>
  <slot>pausePlayback()</slot>
  <slot>normalPlayback()</slot>
  <slot>reversePlayback()</slot>
  <slot>forwardPlayback()</slot>
  <slot>rewindToStartOfPlayback()</slot>
  <slot>forwardToEndOfPlayback()</slot>
  <slot>playbackSetSpeed(double)</slot>
  <slot>setStreamPlaybackTime()</slot>
 </slots>
</ui>
